src/ripple/app/ledger/tests/common_ledger.cpp

```diff
@@ -527,6 +629,32 @@ verifyBalance(Ledger::pointer ledger, TestAccount const& account,
         "balance != amountReq");
 }
 
+void
+verifyLimit (Ledger::pointer ledger, TestAccount const& account,
+    Amount const& amount, std::uint32_t quality_in, std::uint32_t quality_out)
+{
+    auto sle = getLedgerEntryRippleState (ledger, account, amount.getIssuer (),
+        amount.getCurrency ());
+    if (sle == nullptr)
+        throw std::runtime_error ("!sle");
+
+    STAmount limit;
+    amountFromJsonNoThrow (limit, amount.getJson ());
+    if (limit != sle->getFieldAmount (sfHighLimit))
+        throw std::runtime_error("quality_in != HighQualityIn");
```

Comment:
It seems like there's a good chance that this will fail incorrectly if `account` is not the high account in the `sle`. Wouldn't it make sense to check that, then compare `limit` to the appropriate (high or low) limit?